examples/dynamic-lora-sidecar/sidecar/sidecar.py

```diff
@@ -0,0 +1,204 @@
+import requests
+import yaml
+import time
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+import logging
+import datetime
+import os
+
+CONFIG_MAP_FILE = os.environ.get("DYNAMIC_LORA_ROLLOUT_CONFIG", "configmap.yaml")
+DYNAMIC_LORA_FLAG = "VLLM_ALLOW_RUNTIME_LORA_UPDATING"
+BASE_FIELD = "vLLMLoRAConfig"
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+def current_time_human() -> str:
+    now = datetime.datetime.now(datetime.timezone.utc).astimezone()
+    return now.strftime("%Y-%m-%d %H:%M:%S %Z%z")
+
+
+class ConfigWatcher(FileSystemEventHandler):
+    def __init__(self, callback):
+        """
+        Watches config
+
+        Args:
+            callback : Callback function taking no parameters and no return values parsed
+        """
+        self.callback = callback
+
+    def on_modified(self, event):
+        if not event.is_directory and event.src_path.endswith(".yaml"):
+            logging.info(
+                f"Config '{event.src_path}' modified! at '{current_time_human()}'"
+            )
+            self.callback()
+
+
+class LoraReconciler:
+    """
+    Reconciles adapters registered on vllm server with adapters listed in configmap in current state
+    """
+
+    def __init__(self):
+        self.deployment_name = ""
```

Comment:
why don't we just keep the entire config object instead of maintaining each sub-fields? The risk of keep sub-fields is that if you add new fields in the future it's easy to miss it.